public/docs/ts/latest/guide/i18n.jade

```diff
@@ -0,0 +1,302 @@
+    However, the server must also serve translated message bundles back to the application or have tools that embed all
+    translations in the application code at build time.
+    * Our application must track all the pieces of the UI that need to be updated when the locale changes.
+    In addition, if the new language strings are being loaded over the network, this could take time and the UI needs to indicate this in some way to the user.
+    * Allows one to support multiple languages in the same view.
+    As an example, a page could display a table showing how the user's advertising message might look in different locales.
+    This is fairly easy to do with this approach since it's fairly simple to have one locale per root node.
+    * The server is not required to determine the locale from the request – the client side can use cookies, the browser
+    language, and JS APIs to determine the language.
+    However, it's still be beneficial for the server to do some of this (e.g. to serve the likely language pack together with the application)
+    but that can be done at a later stage in development or for production.
+
+  
+a(id="aot")
+.l-main-section
+:marked
+  ### Ahead of time (AoT) compilation approach
+  
+  The AoT approach relies on precompiling the templates using the `ngc` compiler and then injecting the translations into these files.
+  
+  Its main features are:
+    * Performance win: by reloading the entire app, there is no need to track and update the bindings/UI that change as a result of the locale change.
+    * The locale changes so rarely that the cost of the reload is usually incurred only once.
+    * No information/state is (typically) lost. The user is presumably changing the language because he could not understand the earlier language.
+    This means that he doesn't have unsaved information in the application.
+    * Tools support: the strings to translate are easily extracted from our templates.
+    The resulting translations are in a generic format that can be consumed both by Angular and one of the many translation softwares available.
+    * Cannot support multiple languages in the same application view.
+    * We have to reload the app to change the language.
+    * Extra server side support is needed: Since we generate different precompiled files for each language, the server must perform cookie/user agent analysis
+    to decide which localized version of the application code should be returned to the user. This also causes a cache miss.
+    * The server is now responsible for determining the default localized version to serve. (e.g. cookies / geo-ip / Accept-Language header, etc.)
+  
+
+a(id="angular2-i18n")
+.l-main-section
+:marked
+  ## Angular and i18n
+  
+  Whatever approach we decide to use, we must update our templates to define which strings will be translated.
+ 
+  
+a(id="i18n-directive")
+.l-main-section
+:marked
+  ### i18n directive
```

Comment:
`i18n` / `i18n-` are attributes, `<!-- i18n -->` is a comment. may be call them "i18n markers" ?